src/Components/ArtsyShippingEstimate.tsx

```diff
@@ -170,6 +176,104 @@ export const ArtsyShippingEstimate = ({
   )
 }
 
+interface PriceEstimate {
+  destination: string
+  minPrice: number
+  maxPrice?: number | null
+  currency: string
+}
+
+interface UseWidgetObserverProps {
+  onViewEstimatedPrice: (price: PriceEstimate) => void
+}
+
+const useWidgetObserver = ({
+  onViewEstimatedPrice,
+}: UseWidgetObserverProps) => {
+  const widgetObserver = useRef<MutationObserver | null>(null)
+  const [visiblePrice, setVisiblePrice] = useState<PriceEstimate | null>(null)
```

Comment:
This function got so big (mostly about setting up the DOM mutation observer) that i pulled it into its own hook below the component.